src/Mono.Android/PublicAPI/API-34/PublicAPI.Unshipped.txt

```diff
@@ -1 +1,42 @@
-﻿#nullable enable
\ No newline at end of file
+﻿#nullable enable
+abstract Android.Net.Wifi.WifiManager.WpsCallback.OnFailed(Android.Net.Wifi.WpsFailureReason reason) -> void
```

Comment:
This seems bad; this should be `virtual`, so that existing subclasses don't trigger classload asserts because they didn't implement this method.

Didn't we add Metadata support to `generator` to help with this scenario…?

Reply:
We did! `compatVirtualMethod`, added in xamarin/java.interop@53bfb4a72.

Reply:
[On Discord](https://discord.com/channels/732297728826277939/732297837953679412/1205643314620014602), @jpobst noted:

>  it wouldn't have been possible to subclass this type and have it compile, because the JCW wouldn't implement the abstract method, right?  based on that, i'm assuming no one has made a subclass and thus it shouldn't break anyone

Indeed, if we create a `dotnet new android` project, and add `MyCallback.cs` with the following contents:

```csharp
using Android.Net.Wifi;

class MyCallback : WifiManager.WpsCallback {
    public override void OnStarted (string? pin)
    {
        // Do something
    }

    public override void OnSucceeded ()
    {
        // Do something
    }
}
```

The C# code successfully builds, but the Java peer does not:

```
obj/Debug/net8.0-android/android/src/crc64475861335642e0f6/MyCallback.java(4,8):
javac error JAVAC0000:
error: MyCallback is not abstract and does not override abstract method onFailed(int) in WpsCallback
```

Thus, this change is nominally acceptable, as the type could never have been subclassed in the first place.